File: backend/app/schemas/club_management.py
# backend/app/schemas/club_management.py
from datetime import datetime
from sqlmodel import SQLModel

from app.models.club_join_request import JoinRequestStatus
from app.models.user import UserRead
from app.models.book_club_member import MemberRole

class ClubJoinRequestRead(SQLModel):
    id: int
    user_id: int
    book_club_id: int
    status: JoinRequestStatus
    created_at: datetime
    user: UserRead

class ClubMemberRead(SQLModel):
    role: MemberRole
    user: UserRead

class UpdateMemberRoleRequest(SQLModel):
    role: MemberRole

class TransferOwnershipRequest(SQLModel):
    new_owner_id: int

File: frontend/src/types/clubManagement.ts
// frontend/src/types/clubManagement.ts
import type { User } from './auth';

export type MemberRole = 'owner' | 'admin' | 'member';

export interface ClubMember {
  role: MemberRole;
  user: User;
}

export type JoinRequestStatus = 'pending' | 'approved' | 'rejected';

export interface ClubJoinRequest {
  id: number;
  user_id: number;
  book_club_id: number;
  status: JoinRequestStatus;
  created_at: string;
  user: User;
}

// API Request Body Types
export interface BookClubUpdateRequest {
  name?: string;
  description?: string;
  visibility?: 'public' | 'private';
  cover_image_url?: string;
}

export interface UpdateMemberRoleRequest {
  role: MemberRole;
}

export interface TransferOwnershipRequest {
  new_owner_id: number;
}
